File: educational-system-consumer-student-9001/src/main/java/edu/hubu/consumerstudent/service/IStudentCourseService.java
package edu.hubu.consumerstudent.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.student.StudentCourse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "student-provider-service")
public interface IStudentCourseService {

    @GetMapping("/studentCourse/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/studentCourse/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "studentCourseId") String orderBy);

    @GetMapping("/studentCourse/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentCourseId") String orderBy
    );

    @PostMapping("/studentCourse/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentCourseId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.StudentCourse studentCourse
    );

    @PostMapping("/studentCourse/")
    CommonResult save(@RequestBody StudentCourse studentCourse);

    @PutMapping("/studentCourse/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody StudentCourse studentCourse);


    @DeleteMapping("/studentCourse/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/student/IStudentCourseService.kt
package edu.hubu.commons.service.student

import edu.hubu.commons.model.student.StudentCourse;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IStudentCourseService {
    fun save(t: StudentCourse): StudentCourse
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: StudentCourse): StudentCourse
    fun selectOneById(id: String): StudentCourse
    fun selectAllOrderBy(order: String = "studentCourseId"): List<StudentCourse>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentCourseId"): List<StudentCourse>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentCourseId", t: edu.hubu.commons.mybatis.model.StudentCourse): List<edu.hubu.commons.mybatis.model.StudentCourse>
}
